File: house-finder-be/HouseFinder360.RealEstates.Application/RealEstates/Dto/PropertyResponse.cs
using HouseFinder360.RealEstates.Application.Common.Dtos.Shared;

namespace HouseFinder360.RealEstates.Application.RealEstates.Dto;

public class PropertyResponse
{
    public Guid Id { get; set; }
    public string Title { get; set; } = null!;
    public int Price { get; set; }
    public AddressDto Address { get; set; } = null!;
    public string BedroomsNumber { get; set; } = null!;
    public int BathroomsNumber { get; set; }
    public int Area { get; set; }
    public string PropertyType { get; set; } = null!;
    public string Purpose { get; set; } = null!;
    public string? Description { get; set; }
    public IEnumerable<string> PropertyImageUris { get; set; } = null!;
}

File: house-finder-fe/src/shared/model/property-response.ts
import { AddressResponse } from './address.ts'

export interface PropertyResponse {
  id: string
  title: string
  price: number
  address: AddressResponse
  bedroomsNumber: string
  bathroomsNumber: number
  area: 56
  propertyType: string
  purpose: PurposeType
  propertyImageUris: string[]
  description: string
}
type PurposeType = 'Sale' | 'Rent'
